builder/amazon/common/ami_config.go

```diff
@@ -8,23 +8,23 @@ import (
 
 // AMIConfig is for common configuration related to creating AMIs.
 type AMIConfig struct {
-	AMIName                 string            `mapstructure:"ami_name"`
-	AMIDescription          string            `mapstructure:"ami_description"`
-	AMIVirtType             string            `mapstructure:"ami_virtualization_type"`
-	AMIUsers                []string          `mapstructure:"ami_users"`
-	AMIGroups               []string          `mapstructure:"ami_groups"`
-	AMIProductCodes         []string          `mapstructure:"ami_product_codes"`
-	AMIRegions              []string          `mapstructure:"ami_regions"`
-	AMISkipRegionValidation bool              `mapstructure:"skip_region_validation"`
-	AMITags                 map[string]string `mapstructure:"tags"`
-	AMIEnhancedNetworking   bool              `mapstructure:"enhanced_networking"`
```

Comment:
It would be fairly simple to check if `enhanced_networking` is true, but `enhanced_networking_type` is unset, and set it to `sriov`, then you only need to validate the types, and catch the case someone has set the type without setting the bool to true (since it will show as false, rather than unset).

IMO the right behavior is not to imply the bool is true merely by setting type, otherwise this:

```
{
  "enhanced_networking": false,
  "enhanced_networking_type": "sriov"
}
```

Is going to enable it despite an explicit false in the template. So I think we should treat the bool being false with the type defined as an error. That should be done in the builder's Prepare so the config validation is done upfront.